project-minimization-plugin/src/main/kotlin/org/plan/research/minimization/plugin/psi/PsiUtils.kt

```diff
@@ -39,6 +44,42 @@ private typealias ParentChildPsiProcessor<T> = (PsiElement, PsiElement) -> T
  * The PsiProcessor class provides utilities for fetching PSI elements within a given project.
  */
 object PsiUtils {
+    /**
+     * Builds a [PsiStubChildrenCompositionItem] from a [PsiElement].
+     * Should be used with [KtCallExpression] to acquire a trace for a function calls.
+     *
+     * @param context A context with required information
+     * @param element [PsiElement] to build trace to
+     * @return [Option] with a built item
+     * @see PsiStubChildrenCompositionItem
+     */
+    @RequiresReadLock
+    fun buildCompositeStubItem(context: IJDDContext, element: PsiElement) = option {
+        ensure(element !is PsiFile && element is KtElement)
+        val parents = element.parentsWithSelf.toList()
+        val file = parents.last() as? KtFile
+        ensureNotNull(file)
+
+        val (stubPart, childrenPart) = parents.dropLast(1).reversed().splitWhile { KtStub.create(it).isSome() }
+        ensure(stubPart.isNotEmpty())
+
+        val stubs = stubPart.mapNotNull { CompositeIndex.StubDeclarationIndex(KtStub.create(it).bind()) }
```

Comment:
It seems that we are doing the job twice (creation of KtStub)

Reply:
Yep, I suppose it could be rewritten in terms of `canBeCreated` and `create`